### PR TITLE
feat: new hook, useNetworkState

### PR DESCRIPTION
**Pull request by xobotyi**
## What new hook does?

Tracks the state of browser's network connection.

## Checklist

- [x] Have you read [contribution guideline](../../CONTRIBUTING.md)?
- [x] Does the code have comments in hard-to-understand areas?
- [ ] Is there an existing issue for this PR?
  - _link issue here_
- [x] Have the files been linted and formatted?
- [x] Have the docs been updated?
- [x] Have the tests been added to cover new hook?
- [x] Have you run the tests locally to confirm they pass?

### CODE REVIEW

File: src/useNetworkState.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+function getConnectionState(previousState?: IUseNetworkState): IUseNetworkState {
+  const online = navigator?.onLine;
+  const previousOnline = previousState?.online;
+
+  return {
+    online,
+    previous: previousOnline,
+    since: online !== previousOnline ? new Date() : previousState?.since,
+    downlink: conn?.downlink,
```
Comment (JoeDuncko) [inline src/useNetworkState.ts:94]: Why not expose all of `conn` here?

File: stories/useNetworkState.story.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { Example } from './useNetworkState.stories';
+import { Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
+
+<Meta title={'Sensor/useNetwork'} component={Example} />
+
+# useNetwork
+
+Tracks the state of browser's network connection.
+
+> As of the standard, it is not guaranteed that browser connected to the _Internet_, it only
```
Comment (JoeDuncko) [inline stories/useNetworkState.story.mdx:10]: Maybe we should quote MDN here and add a link?

`while you can assume that the browser is offline when it returns a false value, you cannot assume that a true value necessarily means that the browser can access the internet`

https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/onLine

**Review by JoeDuncko — CHANGES_REQUESTED** [review]

File: src/useNetworkState.ts (hunk 1 above)
Comment (xobotyi) [reply src/useNetworkState.ts:94]: `navigator.connection` is a mutable object, also it is EventEmitter instance and it becomes kind of abstraction leak if to simply return it.

Comment (JoeDuncko) [reply src/useNetworkState.ts:94]: Got it

**Pull request merged**